js/interactive-guides/externalized-string.js

```diff
@@ -0,0 +1,182 @@
+var supportedLanguages = ["cs",
+                          "de",
+                          "en",
+                          "es", 
+                          "fr",
+                          "hu",
+                          "it",
+                          "ja",
+                          "ko",
+                          "pl",
+                          "pt-br",
+                          "ro",
+                          "ru",
+                          "zh",
+                          "zh-tw"
+                          ];
+
+/**
+ * Check if browser language is supported by us
+ */
+function checkIfLanguageSupported(browserLanguage){
+    "use strict";
+    var languageCode = "";
+    
+    if (supportedLanguages.indexOf(browserLanguage) !== -1) {
+        languageCode = browserLanguage;
+    } else if (browserLanguage.length === 5) { 
+        var code = browserLanguage.substring(0, 2);
+        if(supportedLanguages.indexOf(code) !== -1) {
+            languageCode = code;
+        }
+    }  
+  
+    return languageCode;
+}
+
+/**
+ * Get language code from the browser's list of languages
+ */
+function getLanguageCode(){
+    "use strict";
+    
+    // Obtain browser languages
+    var browserLanguages = navigator.languages;
+    var languageCode = "";
+    var browserLanguage = "";
+    
+    // Loop through languages and check if supported 
+    if (browserLanguages) {
+        for(var i=0; i<browserLanguages.length; i++){
+            languageCode = checkIfLanguageSupported(browserLanguages[i].toLowerCase());
+            if(languageCode){
+                break;
+            }
+        }
+    } else {
+        browserLanguage = navigator.language ? navigator.language : navigator.userLanguage;
+        languageCode = checkIfLanguageSupported(browserLanguage.toLowerCase());
+    }
+    // Default to English if no language found
+    if (languageCode === ""){
+        languageCode = "en";
+    }
+
+    return languageCode;
+}
+
+/**
+ * Query all the data-externalizedString in html to translate the message.
+ */
+function retrieveExternalizedStrings(rootDir) {
+    'use strict';
+        
+    var languageCode = getLanguageCode();
+
+    var nlsFile = "messages.js";
+    var url = (languageCode !== "en") ? rootDir + languageCode + "/" + nlsFile :  rootDir + nlsFile;
+    
+    //Retrieve translations
+    httpSendRequest(url, replaceExternalizedStrings);  
+}
+
+/**
+ * Send request to the server side base on the browser language to retrieve the translate messages.
+ */
+function httpSendRequest(url, replaceExternalizedStrings) {
+    'use strict';
+    
+    var httpReq = new XMLHttpRequest();
+    var params = null;
+    
+    httpReq.open("GET", url, true);
+
+    // Call a function when the state changes.
+    httpReq.onreadystatechange = function () {
+        if (httpReq.readyState === 4 && httpReq.status === 200) {   
+            replaceExternalizedStrings(httpReq.responseText);
```

Comment:
Does this call to `replaceExternalizedStrings` require the `httpSendRequest` to have it passed into the parameters?

Reply:
no, it doesn't have to pass into as parameter, I removed it